src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java

```diff
@@ -302,6 +310,34 @@ public void validateOptions() throws ConfigurationException
         }
     }
 
+    @Override
+    public void maybeWarnOnOptions()
+    {
+        ImmutableMultimap<String, InetAddressAndPort> dcsNodes = StorageService.instance.getTokenMetadata().getDC2AllEndpoints(snitch);
+        for (Entry<String, String> e : this.configOptions.entrySet())
+        {
+            if (!SchemaConstants.isSystemKeyspace(keyspaceName))
```

Comment:
I think this system keyspace check should be outside of the loop, as int he loop it is causing reader to think why.

Reply:
Right. Left-over from a refactor where ks was indeed changing then but it isn't now.